apps/src/templates/instructions/CommitsAndReviewTab.jsx

```diff
@@ -7,38 +7,60 @@ import Spinner from '@cdo/apps/code-studio/pd/components/spinner';
 import {ViewType} from '@cdo/apps/code-studio/viewAsRedux';
 import CodeReviewDataApi from '@cdo/apps/templates/instructions/codeReviewV2/CodeReviewDataApi';
 import ReviewNavigator from '@cdo/apps/templates/instructions/codeReviewV2/ReviewNavigator';
-import CodeReviewTimelineElement, {
-  codeReviewTimelineElementType
-} from '@cdo/apps/templates/instructions/codeReviewV2/CodeReviewTimelineElement';
-import CodeReviewTimelineCommit from '@cdo/apps/templates/instructions/codeReviewV2/CodeReviewTimelineCommit';
-import CodeReviewTimelineReview from '@cdo/apps/templates/instructions/codeReviewV2/CodeReviewTimelineReview';
+import CodeReviewTimeline from '@cdo/apps/templates/instructions/codeReviewV2/CodeReviewTimeline';
 import Button from '@cdo/apps/templates/Button';
 
 export const VIEWING_CODE_REVIEW_URL_PARAM = 'viewingCodeReview';
 
-const CommitsAndReviewTab = ({
-  onLoadComplete,
-  channelId,
-  serverLevelId,
-  serverScriptId,
-  viewAsCodeReviewer,
-  viewAsTeacher,
-  userIsTeacher,
-  codeReviewEnabled
-}) => {
-  const [loadingReviewData, setLoadingReviewData] = useState(false);
-
-  const dataApi = new CodeReviewDataApi(
+const CommitsAndReviewTab = props => {
+  const {
+    onLoadComplete,
     channelId,
     serverLevelId,
-    serverScriptId
+    serverScriptId,
+    viewAsCodeReviewer,
+    viewAsTeacher,
+    userIsTeacher,
+    codeReviewEnabled
+  } = props;
+
+  const [loadingReviewData, setLoadingReviewData] = useState(false);
+  const [reviewData, setReviewData] = useState([]);
+  const [commitsData, setCommitsData] = useState([]);
+  const [hasOpenCodeReview, setHasOpenCodeReview] = useState(null);
+
+  const dataApi = useMemo(
```

Comment:
This interaction between dataApi, refresh and the useEffect seems a bit complicated.

I think `useMemo` is intended as an optimization -- React is allowed to throw away the memoized value and recalculate if it wants to.  Unfortunately, I think that would lead to a refresh here which seems undesirable.  I think my suggestion here is to not have the data api be instanced.  Just use it to organize some exports and pass all of the needed parameters to getCodeReviews and getCommits.

I'm not sure about this, but it might also be helpful to have the data api expose a single method (hook?) that gets both code reviews and commits and combines that data (the logic that's currently in CodeReviewTimeline).  One way to think about this is that the fact this data is coming from two separate APIs is abstracted from all of the UX components.  This might then simplify refresh enough that we don't need to define it separately and can call the data api directly from the useEffect.

Finally, just throwing it out there for the future, I wonder if useFetch would help further simplify some of the logic.

(Sorry, this ended up being a lot.  Also happy to chat in person.)

Reply:
One of the benefits of having an instance is storing the csrf token, which could be stored on the component instead. I do see your point about having some of the combining logic extracted. I'll play around with it. Things actually change a bit in the next PR where I decide to store the open code review separately on the state so that I can more easily mutate it when changes are made: https://github.com/code-dot-org/code-dot-org/pull/46150

Reply:
Sounds good, up to you whether you want to continue to iterate on this PR or if you want to merge this and come back to it later.

Reply:
I'm looking at potential refactors on the next branch, so I'm going to merge this one and keep looking at it.